src/Concordium/Client/GRPC2.hs

```diff
@@ -2359,42 +2360,343 @@ instance ToProto SendBlockItemInput where
+-- in the reward period containing the given block.
+getPoolDelegatorsRewardPeriodV2 :: (MonadIO m) => BlockHashInput -> BakerId -> ClientMonad m (GRPCResult (Maybe (Seq.Seq QueryTypes.DelegatorRewardPeriodInfo)))
+getPoolDelegatorsRewardPeriodV2 bhInput baker =
+    withServerStreamCollectV2 (callV2 @"getPoolDelegatorsRewardPeriod") msg ((fmap . fmap . mapM) fromProto)
+  where
+    msg = defMessage & ProtoFields.blockHash .~ toProto bhInput & ProtoFields.baker .~ toProto baker
+
+-- |Get all registered delegators of a given pool at the end of a given block.
+getPoolDelegatorsV2 :: (MonadIO m) => BlockHashInput -> BakerId -> ClientMonad m (GRPCResult (Maybe (Seq.Seq QueryTypes.DelegatorInfo)))
+getPoolDelegatorsV2 bhInput baker =
+    withServerStreamCollectV2 (callV2 @"getPoolDelegators") msg ((fmap . fmap . mapM) fromProto)
+  where
+    msg = defMessage & ProtoFields.blockHash .~ toProto bhInput & ProtoFields.baker .~ toProto baker
+
+-- |Get IDs of all bakers at the end of a given block.
+getBakerListV2 :: (MonadIO m) => BlockHashInput -> ClientMonad m (GRPCResult (Maybe (Seq.Seq BakerId)))
+getBakerListV2 bhInput = withServerStreamCollectV2 (callV2 @"getBakerList") msg ((fmap . fmap . mapM) fromProto)
+  where
+    msg = toProto bhInput
+
+-- |Get key-value pairs representing the entire state of a specific contract instance in a given block.
+-- The resulting sequence is ordered lexicographically according to its keys.
+getInstanceStateV2 :: (MonadIO m) => BlockHashInput -> ContractAddress -> ClientMonad m (GRPCResult (Maybe (Seq.Seq (ByteString, ByteString))))
+getInstanceStateV2 bhInput cAddress =
+    withServerStreamCollectV2 (callV2 @"getInstanceState") msg ((fmap . fmap . mapM) fromProto)
+  where
+    msg = defMessage & ProtoFields.blockHash .~ toProto bhInput & ProtoFields.address .~ toProto cAddress
+
+-- |Get the addresses of all smart contract instances in a given block.
+getInstanceListV2 :: (MonadIO m) => BlockHashInput -> ClientMonad m (GRPCResult (Maybe (Seq.Seq ContractAddress)))
+getInstanceListV2 bhInput = withServerStreamCollectV2 (callV2 @"getInstanceList") msg ((fmap . fmap . mapM) fromProto)
+  where
+    msg = toProto bhInput
+
+-- |Get ancestors of a given block.
+-- The first element of the sequence is the requested block itself, and the block
+-- immediately following a block in the sequence is the parent of that block.
+-- The sequence contains at most `limit` blocks, and if the sequence is
+-- strictly shorter, the last block in the list is the genesis block.
+getAncestorsV2 :: (MonadIO m) => BlockHashInput -> Word64 -> ClientMonad m (GRPCResult (Maybe (Seq.Seq BlockHash)))
+getAncestorsV2 bhInput limit = withServerStreamCollectV2 (callV2 @"getAncestors") msg ((fmap . fmap . mapM) fromProto)
+  where
+    msg = 
+        defMessage
+            & ProtoFields.blockHash .~ toProto bhInput
+            & ProtoFields.amount .~ limit
+
+-- |Get all smart contract modules in a given block.
```

Comment:
The phrasing used for `getAccountListV2` is more precise and clear. This sounds like it only will return modules from this specific block.

Reply:
Thanks.